new-log-viewer/src/services/LogFileManager/index.ts

```diff
@@ -254,6 +276,84 @@ class LogFileManager {
         };
     }
 
+    /**
+     * Searches for log events based on the given search string.
+     *
+     * @param searchString
+     * @param isRegex
+     * @param isCaseSensitive
+     */
+    startQuery (searchString: string, isRegex: boolean, isCaseSensitive: boolean): void {
+        this.#queryId++;
+
+        // If the search string is empty, or there are no logs, return
+        if ("" === searchString) {
+            return;
+        } else if (0 === this.#numEvents) {
+            return;
+        }
```

Comment:
It might look cleaner this way
```suggestion
        if ("" === searchString || 0 === this.#numEvents) {
            return;
        }
```